### PR TITLE
MWPW-121476 call interlinks

### PR DESCRIPTION
**Pull request by vhargrave**
Fixes the interlinks. The issue was that we were importing the interlink module, but not calling its init method. 

Resolves: [MWPW-121476](https://jira.corp.adobe.com/browse/MWPW-121476)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/amauchley/interlinks
- After: [https://vhargrave-interlinks--milo--adobecom.hlx.page/?martech=off](https://vhargrave-interlinks--milo--adobecom.hlx.page/drafts/amauchley/interlinks)

### CODE REVIEW

**Review by chrischrischris — APPROVED** [review]

File: libs/scripts/scripts.js [hunk 1]
```diff
@@ -117,5 +117,7 @@ const config = {
 (async function loadPage() {
   setConfig(config);
   await loadArea();
-  loadDelayed();
+  loadDelayed().then((interlinksMod) => {
+    if (interlinksMod) interlinksMod.default('/keywords.json');
```
Comment (chrischrischris) [inline libs/scripts/scripts.js:121]: Can we do this inside of `loadDelayed` instead?  I'd prefer to keep this as clean as possible.

**Review by chrischrischris — CHANGES_REQUESTED** [review]

File: libs/features/interlinks.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ export function checkAndAddMatch(matches, contender, maxMatches) {
  * @param {string} path The location of the keywords file to be used for interlinks.
  * @param {number} limit The maximum amount of keywords to fetch from the file.  Default is 1000.
  */
-export default async function interlink(path, limit = 1000) {
+export default async function interlink(path = '/keywords.json', limit = 1000) {
```
Comment (auniverseaway) [inline libs/features/interlinks.js:26]: You should be using `${config.locale.contentRoot}/keywords.json`;

File: libs/utils/utils.js [hunk 3]
```diff
@@ -550,14 +550,12 @@ export function loadDelayed(delay = 3000) {
...
       if (getMetadata('interlinks') === 'on') {
-        import('../features/interlinks.js').then((mod) => {
-          resolve(mod);
-        });
-      } else {
-        resolve(null);
+        promises.push(import('../features/interlinks.js').then((mod) => mod.default()));
       }
-      import('./samplerum.js').then(({ sampleRUM }) => sampleRUM('cwv'));
+      promises.push(import('./samplerum.js').then(({ sampleRUM }) => sampleRUM('cwv')));
```
[the remark this answers is not in the document]
Comment (auniverseaway) [reply libs/utils/utils.js:557]: I think this is overly complicated for what we actually need. This function runs 3s after the last important part of the page. There's no reason to have all the complexity of a promise array.

If I remember correctly, I turned this into a promise for easier testing failure, but I think these new additions are overkill.

**Review by auniverseaway — CHANGES_REQUESTED** [review]

File: libs/utils/utils.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (vhargrave) [reply libs/utils/utils.js]: ok, I had started thinking that some milo-college projects were waiting for the delayed method to then do their own stuff and so I wanted the delayed method to at least be accurate with its response. Thanks for looking! I've reverted my changes.

File: libs/features/interlinks.js (hunk 2 above)
Comment (vhargrave) [reply libs/features/interlinks.js]: I keep getting caught out by that configRoot don't I?

File: libs/features/interlinks.js [hunk 4]
```diff
@@ -23,7 +23,7 @@ export function checkAndAddMatch(matches, contender, maxMatches) {
  * @param {string} path The location of the keywords file to be used for interlinks.
  * @param {number} limit The maximum amount of keywords to fetch from the file.  Default is 1000.
  */
-export default async function interlink(path, limit = 1000) {
+export default async function interlink(config, path = `${config.locale.contentRoot}/keywords.json`, limit = 1000) {
   const articleBody = document.querySelector('main');
```
Comment (auniverseaway) [inline libs/features/interlinks.js:27]: Can't you avoid a lot of this hassle by simply passing in the path from utils? Then you don't have to:

1. Change every single test.
2. Pass in the config object to interlinks
3. Set a default path in the function

**Review by auniverseaway — CHANGES_REQUESTED** [review]

File: libs/features/interlinks.js (hunk 4 above)
Comment (vhargrave) [reply libs/features/interlinks.js:27]: I think we're going to need the config for later tasks: 
- https://jira.corp.adobe.com/browse/MWPW-121870
- https://jira.corp.adobe.com/browse/MWPW-121872

It's not necessary this task though. I'll change it

**Review by auniverseaway — APPROVED** [review]
@chrischrischris can you re-review? I think we're good now.

**Review by chrischrischris — APPROVED** [review]

**Comment by auniverseaway** [comment]
@vhargrave you have merge conflicts.

**Comment by vhargrave** [comment]
> @vhargrave you have merge conflicts.

@auniverseaway fixed!

**Pull request merged**